src/Microsoft.Graph.Cli.Core/Authentication/AuthenticationServiceFactory.cs

```diff
@@ -41,12 +41,13 @@ public AuthenticationServiceFactory(IPathUtility pathUtility, IAuthenticationCac
     /// <param name="clientId">Client Id</param>
     /// <param name="certificateName">Certificate name</param>
     /// <param name="certificateThumbPrint">Certificate thumb-print</param>
+    /// <param name="environment">The national cloud environment. Either 'Global', 'US_GOV', 'US_GOV_DOD' or 'China'</param>
     /// <param name="cancellationToken">Cancellation token</param>
     /// <returns>Returns a login service instance.</returns>
     /// <exception cref="InvalidOperationException">When an unsupported authentication strategy is provided.</exception>
-    public virtual async Task<LoginServiceBase> GetAuthenticationServiceAsync(AuthenticationStrategy strategy, string? tenantId, string? clientId, string? certificateName, string? certificateThumbPrint, CancellationToken cancellationToken = default)
+    public virtual async Task<LoginServiceBase> GetAuthenticationServiceAsync(AuthenticationStrategy strategy, string? tenantId, string? clientId, string? certificateName, string? certificateThumbPrint, CloudEnvironment environment, CancellationToken cancellationToken = default)
```

Comment:
since the parameter is defaulted and optional in the GetTokenCredentialAsync method, should it be here as well for consistency of the API surface?

Reply:
Being defaulted is actually a mistake. I'll remove the default. Thanks for catching this.